### PR TITLE
Lt/view song preview

### PR DESCRIPTION
**Pull request by leethomas**
Incomplete, just PRing to get an early review and for @josephli036 to take a look. Clicking on a row in the table loads the side view w/ lyrics.

Looks like this right now: .

Should look something like this: https://app.moqups.com/leet/aSVpyORnR2/view/page/a9d37a182

Still need to:
 * ~~move JS to it's own file~~
 * ~~show key & tempo~~
 * display keywords that were not found <--- Not doing this for now, going for MVP.
 * ~~hide/show side drawer (it's always there right now)~~
 * ~~action buttons for editing & printer friendly view?~~

### CODE REVIEW

File: app/controllers/songs_controller.rb [hunk 1]
```diff
@@ -31,6 +32,16 @@ def index
     end
   end
 
+  def show
+    respond_to do |format|
+      @song = Song.find(params[:id])
+      song = @song
```
Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: `song` doesn't seem to be used again. And why do we need to make `@song` an instance variable if we are rendering it as JSON directly and not going into a erb view?

**Comment by natyconnor** [comment]
Love the sidebar. Perfect for not losing context and lends itself naturally to going full screen and back. 

However, I also want to be weary with throwing in tons of js into the view. I know js in rails is kind of messy, and the asset pipeline can be a pain, but maintaining in-line js a few years from now can be even worse. @patrickf3139 what are your thoughts on organizing js in Rails?

**Comment by leethomas** [comment]
Thanks! I agree, we should move the JS to it's own file. I'll add it to the to-do list in the first post.

**Comment by patrickf3139** [comment]
@natyconnor Good catch! Yeah, I think it belongs in assets.

**Comment by leethomas** [comment]
going to finish this up today

**Comment by patrickf3139** [comment]
Alright! Looking forward to it.

**Comment by leethomas** [comment]
Ready for review again. Screenshot updated

File: app/assets/javascripts/songs/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  var populateDrawer = function(song) {
+    var drawer = $('.preview-drawer');
+    drawer.find('.name').text(song.name);
+    drawer.find('.song-sheet').html(song.chordSheet);
+    drawer.find('.tempo').text('Tempo: ' + song.tempo);
+    drawer.find('.key').text('Key: ' + song.key);
+    drawer.find('.actions .edit').attr('href', song.editPath);
+  }
```
Comment (patrickf3139) [inline app/assets/javascripts/songs/index.js]: Can you also show the artist field? It's a pretty prominent column in the table after all.

File: app/assets/javascripts/songs/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    $('.songs-table .selected').removeClass('selected');
+    $(this).addClass('selected');
+
+    $.get(url, function (data) {
+      var song = data.song;
+      var chordSheet = song.chord_sheet;
+      var keywords = $('#songs-search-field').val();
+
+      if (keywords !== '') {
+        chordSheet = highlightedSongSheet(keywords, chordSheet);
```
Comment (patrickf3139) [inline app/assets/javascripts/songs/index.js]: Can we also do the highlighting for the title and artist field?

**Comment by leethomas** [comment]
All comments addressed, sorry about that! Ready for review once more and screenshot updated again.

**Comment by patrickf3139** [comment]
Dude, Lee, wonderful job! Thanks for your hard work. LGTM. Optional change if you like the idea and have time: omit artist, tempo, key if they are blank rather than showing N/A. But I don't mind at all if you just merge because it's already pretty awesome.

File: app/assets/javascripts/songs/index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+  // hide drawer if anything outside of the table & drawer are clicked
+  $('body').click(function(e) {
+    var drawer = $('.preview-drawer');
+    var target = $(e.target);
+
+    if (!target.parents('.songs-table').length &&
+      !target.parents('.preview-drawer').length && !target.is(drawer)) {
+      hideDrawer();
+    }
```
Comment (patrickf3139) [inline app/assets/javascripts/songs/index.js]: This `hidesDrawer()` when the user clicks on what exactly? Just curious.

**Comment by patrickf3139** [comment]
Also, I noticed a small bug where the bottom of the lyrics is cut off on longer songs (you can try Glorious Day by Casting Crowns to repo). Also not a big deal and we can fix it later.

File: app/assets/javascripts/songs/index.js (hunk 4 above)
Comment (leethomas) [inline app/assets/javascripts/songs/index.js]: Anywhere outside of the table or the drawer itself

**Comment by leethomas** [comment]
Thanks Patrick! Merging now. I'll look into the cut off bug later. I made the change to omit info when it isn't there.

**Comment by patrickf3139** [comment]
Oh yeah! I'll deploy this later today.

**Pull request merged**